src/java.base/share/classes/java/text/DecimalFormat.java

```diff
@@ -449,6 +449,13 @@ public DecimalFormat() {
      * for the default {@link java.util.Locale.Category#FORMAT FORMAT} locale.
      * This is a convenient way to obtain a
      * DecimalFormat when internationalization is not the main concern.
+     * If the given pattern is in scientific notation, the limit of integer
+     * digits will be derived from the pattern. Otherwise, if the pattern is not
+     * in scientific notation, the limit of integer digits will not be based on the
+     * pattern, and instead set to {@link Integer#MAX_VALUE}. This behavior
+     * is the typical end-user desire. Use {@link #setMaximumIntegerDigits(int)}
+     * if you want to set a specific limit to integer digits when the
+     * given pattern is not in scientific notation.
```

Comment:
It could be simpler if the normal case (Integer.MAX_VALUE) comes first, then the scientific notation case in which you can refer to the max integer digit in the class description. Also, `setMaximumIntegerDigits` can be called on a scientific notation as well, correct?

Reply:
Thank you for the review. Will adjust with your suggestions.

> Also, `setMaximumIntegerDigits` can be called on a scientific notation as well, correct?

I meant that you pretty much only need to consider calling `setMaximumIntegerDigits` when the pattern is not in scientific notation, not that you are not able to. Will make this clearer.

Reply:
@naotoj Updated with your suggestions. Held off on updating CSR as well in case the wording still needs to be further adjusted. I also considered adding these changes to the class description (instead of each method/constructor), but I thought the class description was already quite long, and it could easily be missed if added that way.